bundler/spec/runtime/setup_spec.rb

```diff
@@ -1072,6 +1077,10 @@ def clean_load_path(lp)
     end
 
     it "evals each gemspec with a binding from the top level" do
+      gemfile <<-G
+        source "#{file_uri_for(gem_repo1)}"
+      G
+
```

Comment:
I think you're already "fixing" the Gemfile in the `before` hook, so you don't need to create a separate `Gemfile` here?